### DISCUSSION TITLE
New Donation Flow - Authentication Discussion

### DISCUSSION THREAD

**Discussion by dimitur2204** [podkrepi-bg/frontend#1298] [General]
This is a continuation of the discussion in https://github.com/podkrepi-bg/frontend/pull/1294#discussion_r1083538358

This is what the flow currently looks like for the Anonymous and Not registered users.

Not logged|Logged in
---|---
<img width="785" alt="image" src="https://user-images.githubusercontent.com/61479393/214136856-aed9ece3-3e4f-4aa7-9193-6db6fb422455.png">|<img width="770" alt="image" src="https://user-images.githubusercontent.com/61479393/214137016-88e4f83a-83c2-4cd6-98fc-4fba9fcab208.png">

What exactly do we want to happen?

## Ideas:
* A checkbox changing depending on that if you are logged in or not to "I want to be anonymous" or "Continue without registration"
* Remove the email field from the new Payment Information form (see the gif below) and add a third radio ("I want to be anonymous" or "Continue without registration") button that makes an email field pop up.
* Other ideas?

New flow latest update:
![full-flow-after-auth](https://user-images.githubusercontent.com/61479393/214137838-7c559d9e-3476-4af7-a923-0bd85ba19f1e.gif)

**Comment by dimitur2204** [comment]
@kachar @Juicebgs @swolf86 
Created this discussion as continuation of what we had on the PR since that is not the place to do it and it's still unclear to be what exactly do we want to happen

**Comment by dimitur2204** [comment]
Does this latest change down here I did with the checkboxes look good and is it understandable to the user, that his email will actually be used and is needed

Anon|Logged
---|---
![anon-auth-v2](https://user-images.githubusercontent.com/61479393/214142212-b669df9d-05f7-46b2-84c6-45143a9c9b41.gif)|![logged-auth-v2](https://user-images.githubusercontent.com/61479393/214142244-7e10660c-9011-4efc-a211-dd78c8889878.gif)

**Reply by kachar** [reply]
I think the user should be able to choose between the following options:

- user is authenticated
  - skip to the next section
- user is not authenticated
  - :o:  login
  - :o: register
  - :o: continue without registration

Additionally, the user should be given the opportunity to **donate anonymously** as a checkbox seen at all times (auth/no auth):

- :black_square_button:   donate anonymously

---

In the flow you've captured it seems the option "Continue without registration" is what needs to happen with the detail that currently it does not hide the section.
